swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -136,20 +132,15 @@ class Adapter {
           logFilter,
           callbackHandler
         )
-      // Update our internal state
-      self.state = .tunnelStarted(session: session)
 
       // Start listening for network change events. The first few will be our
       // tunnel interface coming up, but that's ok -- it will trigger a `set_dns`
       // connlib.
       beginPathMonitoring()
 
-      // Tell the system the tunnel is up, moving the tunnelManager status to
-      // `connected`.
-      completionHandler(nil)
+      // Update state in case everything succeeded
+      self.state = .tunnelStarted(session: session)
     } catch let error {
```

Comment:
I wanted to avoid this `catch` altogether but I couldn't find a concise way of telling Swift that it should convert this error to an `AdapterError`.

In Rust, I would write `WrappedSession::connect(..).map_err(AdapterError::connlib_connect_error)?` but there doesn't seem to be an equivalent of that?